haskell_scotty/Server.hs

```diff
@@ -22,4 +22,9 @@ main = scotty 6969 $ do
     jsonAPI client
     status status201
 
+  delete "/api/clients/:id" $ do
+    clientId <- param "id"
+    liftIO $ deleteClient clientId
```

Comment:
Ah, that makes total sense. Thanks for the explanation!

I wasn't suggesting that we actually desugar the `do` syntax, I just thought that maybe the reason for needing to `liftIO` might be more obvious if it were desugared, for instructional purposes. But it sounds like the answer is "no, Micah just needs to level up his understanding of monads." :)